code/game/dna/genes/vg_powers.dm

```diff
@@ -46,82 +46,119 @@
 		if(new_head_accessory)
-			head_organ.ha_style = new_head_accessory
+			M.change_head_accessory(new_head_accessory)
 
 		var/new_head_accessory_colour = input("Please select head accessory colour.", "Character Generation", rgb(head_organ.r_headacc, head_organ.g_headacc, head_organ.b_headacc)) as null|color
 		if(new_head_accessory_colour)
-			head_organ.r_headacc = hex2num(copytext(new_head_accessory_colour, 2, 4))
-			head_organ.g_headacc = hex2num(copytext(new_head_accessory_colour, 4, 6))
-			head_organ.b_headacc = hex2num(copytext(new_head_accessory_colour, 6, 8))
-
-	//Body markings.
-	if(M.species.bodyflags & HAS_MARKINGS)
-		var/list/valid_markings = M.generate_valid_markings()
-		var/new_marking = input("Please select marking style", "Character Generation", M.m_style) as null|anything in valid_markings
-		if(new_marking)
-			M.m_style = new_marking
-
-		var/new_marking_colour = input("Please select marking colour.", "Character Generation", rgb(M.r_markings, M.g_markings, M.b_markings)) as null|color
-		if(new_marking_colour)
-			M.r_markings = hex2num(copytext(new_marking_colour, 2, 4))
-			M.g_markings = hex2num(copytext(new_marking_colour, 4, 6))
-			M.b_markings = hex2num(copytext(new_marking_colour, 6, 8))
+			M.change_head_accessory_color(hex2num(copytext(new_head_accessory_colour, 2, 4)), hex2num(copytext(new_head_accessory_colour, 4, 6)), hex2num(copytext(new_head_accessory_colour, 6, 8)))
 
 	//Body accessory.
 	if(M.species.tail && M.species.bodyflags & HAS_TAIL)
 		var/list/valid_body_accessories = M.generate_valid_body_accessories()
 		if(valid_body_accessories.len > 1) //By default valid_body_accessories will always have at the very least a 'none' entry populating the list, even if the user's species is not present in any of the list items.
 			var/new_body_accessory = input("Please select body accessory style", "Character Generation", M.body_accessory) as null|anything in valid_body_accessories
 			if(new_body_accessory)
-				M.body_accessory = body_accessory_by_name[new_body_accessory]
+				M.change_body_accessory(new_body_accessory)
+
+	//Head markings.
+	if(M.species.bodyflags & HAS_HEAD_MARKINGS)
+		var/list/marking_styles = params2list(M.m_styles)
+		var/list/valid_head_markings = M.generate_valid_markings("head")
+		var/new_marking = input("Please select head marking style", "Character Generation", marking_styles["head"]) as null|anything in valid_head_markings
+		if(new_marking)
+			M.change_markings(new_marking, "head")
+
+		var/list/marking_colours = params2list(M.m_colours)
+		marking_colours["head"] = sanitize_hexcolor(marking_colours["head"])
+		var/new_marking_colour = input("Please select head marking colour.", "Character Generation", rgb(hex2num(copytext(marking_colours["head"], 2, 4)), hex2num(copytext(marking_colours["head"], 4, 6)), hex2num(copytext(marking_colours["head"], 6, 8)))) as null|color
```

Comment:
This takes a hex color, converts it to separate R/G/B values, then converts it back to a hex color. Why.